src/nodes/ListItem.js

```diff
@@ -128,12 +128,14 @@ export default class ListItem extends TiptapListItem {
 						return false
 					}
 
-					tr.setNodeMarkup(parentList.pos, schema.nodes.list_item, { type: parentList.node.attrs.type === TYPES.CHECKBOX ? TYPES.BULLET : TYPES.CHECKBOX })
+					state.doc.nodesBetween(selection.from, selection.to, (node, pos) => {
```

Comment:
```suggestion
					tr.doc.nodesBetween(tr.selection.from, tr.selection.to, (node, pos) => {
```

Shouldn't this operate on the transaction instead of the original document?

Reply:
With that it would also fix #903 right away it seems, otherwise when you select a block of paragraph nodes and trigger the checklist, they will only get converted to a regular bullet list, as the `toggleList` call from above is not yet dispatched to the original `state.doc`